### PR TITLE
message forwarding UI

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

File: shared/chat/conversation/fwd-msg/team-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+type Props = Container.RouteProps<{srcConvID: Types.ConversationIDKey; msgID: Types.MessageID}>
+
+const TeamPicker = (props: Props) => {
+  const srcConvID = Types.keyToConversationID(Container.getRouteProps(props, 'srcConvID', ''))
+  const msgID = Container.getRouteProps(props, 'msgID', 0)
+  const [term, setTerm] = React.useState('')
+  const [results, setResults] = React.useState<Array<RPCChatTypes.AddBotConvSearchHit>>([])
+  const [waiting, setWaiting] = React.useState(false)
+  const [error, setError] = React.useState('')
+  const fwdMsg = Container.useRPC(RPCChatTypes.localForwardMessageRpcPromise)
```
Comment (joshblum) [inline shared/chat/conversation/fwd-msg/team-picker.tsx:22]: TODO: get nonblock working

File: shared/chat/conversation/fwd-msg/team-picker.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  React.useEffect(() => {
+    doSearch()
+  }, [doSearch])
+
+  const renderResult = (index: number, item: RPCChatTypes.AddBotConvSearchHit) => {
+    return (
+      <Kb.ClickableBox key={index} onClick={() => onSelect(item.convID)}>
+        <Kb.Box2 direction="horizontal" fullWidth={true} gap="tiny" style={styles.results}>
+          {item.isTeam ? (
+            <TeamAvatar isHovered={false} isMuted={false} isSelected={false} teamname={item.name} />
```
Comment (hyperobject) [inline shared/chat/conversation/fwd-msg/team-picker.tsx:88]: a little unclear what this does with a big team, might be good to append '#general' if we're only supporting forwarding to the default convo for teams

File: shared/chat/conversation/fwd-msg/team-picker.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+          paddingBottom: Styles.globalMargins.tiny,
+        },
+      }),
+      searchFilter: Styles.platformStyles({
+        common: {
+          marginBottom: Styles.globalMargins.xsmall,
+          marginTop: Styles.globalMargins.tiny,
+        },
+        isElectron: {
+          marginLeft: Styles.globalMargins.small,
```
[the remark this answers is not in the document]
Comment (joshblum) [reply shared/chat/conversation/fwd-msg/team-picker.tsx:174]: this is verbatim from the bot picker, should i change both?

File: shared/chat/conversation/fwd-msg/team-picker.tsx (hunk 2 above)
Comment (joshblum) [reply shared/chat/conversation/fwd-msg/team-picker.tsx:88]: oops this should support channels too

**Comment by joshblum** [comment]
> permission to post in a team before showing it in the picker

you mean min writer role?

File: shared/chat/conversation/fwd-msg/team-picker.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (hyperobject) [reply shared/chat/conversation/fwd-msg/team-picker.tsx:174]: oh hmm maybe? it looks weird to me but might be good to check with design

**Comment by hyperobject** [comment]
> > permission to post in a team before showing it in the picker
> 
> you mean min writer role?

yeah, a team where the min writer role was set to higher than my role showed up and I was able to select it. it showed a reasonable error in the conversation but imo it'd be better to just not show in the pocket

File: shared/chat/conversation/fwd-msg/team-picker.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshblum) [reply shared/chat/conversation/fwd-msg/team-picker.tsx:174]: cc @adamjspooner

File: shared/chat/conversation/fwd-msg/team-picker.tsx [hunk 4]
```diff
@@ -113,7 +118,7 @@ const TeamPicker = (props: Props) => {
         ) : (
           undefined
         ),
-        title: 'Add to team or chat',
+        title: 'Forward team or chat',
```
Comment (hyperobject) [inline shared/chat/conversation/fwd-msg/team-picker.tsx:121]: "forward to" might be a little clearer here

File: go/chat/server.go [hunk 5]
```diff
@@ -3822,3 +3827,200 @@ func (h *Server) ToggleEmojiAnimations(ctx context.Context, enabled bool) (err e
...
+				IdentifyBehavior:  arg.IdentifyBehavior,
+				EphemeralLifetime: ephemeralLifetime,
+			},
+		})
+		if err != nil {
+			return res, err
+		}
+		go func() {
+			defer os.Remove(dl.FilePath)
+			outbox := storage.NewOutbox(h.G(), uid)
```
Comment (mmaxim) [inline go/chat/server.go:3961]: Use GetUploadTempFile

Comment (joshblum) [reply go/chat/server.go:3961]: @mmaxim not sure how that applies here, what do you mean exactly? we download the file first and then just cleanup in this routine.

**Comment by joshblum** [comment]
@mmaxim this is ready

**Pull request merged**